backend/games/word_guesser/WordGuesser_test.py

```diff
@@ -0,0 +1,73 @@
+import pytest
+
+from backend.games.word_guesser.WordGuesser import WordGuesser
+
+MAX_GUESSES = 6
+WORD_LIST = [
+    "adieu",
+    "odium",
+    "shade",
+    "resin",
+    "alert",
+    "haunt",
+    "orate",
+    "media",
+    "blind",
+    "route",
+    "audio",
+    "pause",
+    "alien",
+    "canoe",
+    "plane",
+    "rouse",
+    "fraud",
+    "atone",
+    "raise",
+    "minor",
+]
+
+# basic expected behavior
+def test_constructor():
+    game = WordGuesser()
+
+    assert game.goal_word == ""
+
+
+def test_start_game():
+    game = WordGuesser()
+    game.start_game()
+    word = game.goal_word
+    assert word in WORD_LIST
+
+
+def test_guesses():
+    game = WordGuesser()
+    game.start_game()
+    word = game.goal_word
+
+    game.guess_word("minor")
+    tmp_guess = {
+        "guess": "minor",
+        "results": [
+            "correct",
+            "not_in_word",
+            "not_in_word",
+            "in_word",
+            "not_in_word",
+        ],
+    }
+    assert game.guesses["guess"] == tmp_guess["guess"]
```

Comment:
How does this test case pass? If the goal word is chosen randomly, how do you make sure the guess `"minor"` has the same results?

Ideally, you would have the process for choosing a word separate from the game logic, that way you do not have randomness in a component that you are testing for deterministic logic. And you can test the random word chooser separately, even though it is random, it will adhere to some testable properties, like how you asserted that the goal word is in the list of options.

The next best thing is to be able to override the goal word when instantiating the game class. If a goal word is specified in the constructor, then use that, otherwise pick one at random.

This will allow you to write deterministic test cases for your game class and test specific edge cases.

Reply:
- changed so that the default goal word is "basic" 
- start_game() is the function that randomizes the goal word, so it is not called in the tests
- start_game() is tested separately in test_start_game()